### PR TITLE
Add tasks to user configuration

### PR DESCRIPTION
**Pull request by alexr00**
Part of https://github.com/microsoft/vscode/issues/1435

### CODE REVIEW

**Comment by sandy081** [comment]
@alexr00 After more thinking,  we shall not exclude loading tasks and launches during initialisation. Because it's contract that extensions receives the complete configuration initially.

It means you can replace `UserConfiguration` with `FileServiceBasedUserConfiguration`. In fact I would do it other way around. Rename `FileServiceBasedUserConfiguration` with `UserConfiguration`. 

Also since `FileServiceBasedFolderConfiguration` and `FileServiceBasedUserConfiguration` are doing the same, is it possible to reuse or make it to a generic one?

**Comment by alexr00** [comment]
@sandy081 I added some tests, is there any other test suite I should add to?

File: src/vs/workbench/services/configuration/browser/configuration.ts [hunk 1]
```diff
@@ -51,17 +52,30 @@ export class UserConfiguration extends Disposable {
...
 		this.reloadConfigurationScheduler = this._register(new RunOnceScheduler(() => this.reload().then(configurationModel => this._onDidChangeConfiguration.fire(configurationModel)), 50));
 		this._register(Event.filter(this.fileService.onFileChanges, e => e.contains(this.userSettingsResource))(() => this.reloadConfigurationScheduler.schedule()));
+		runWhenIdle(() => {
+			this._userConfiguration = this.createAndRegisterConfiguration();
+			this.reload().then(configurationModel => this._onDidChangeConfiguration.fire(configurationModel));
+		}, 5000);
+	}
+
+	private createAndRegisterConfiguration(): FileServiceBasedUserConfiguration {
+		const userSettingsFolder = resources.dirname(this.userSettingsResource.with({ scheme: Schemas.file }));
```
Comment (sandy081) [inline src/vs/workbench/services/configuration/browser/configuration.ts]: Need not to change scheme to file

File: src/vs/workbench/services/configuration/browser/configuration.ts [hunk 2]
```diff
@@ -51,17 +52,30 @@ export class UserConfiguration extends Disposable {
...
+		const userConfiguration = new FileServiceBasedUserConfiguration(userSettingsFolder, [FOLDER_SETTINGS_NAME, TASKS_CONFIGURATION_KEY], this.scopes, this.fileService);
+		this._register(userConfiguration.onDidChange(e => this.reload().then(configurationModel => this._onDidChangeConfiguration.fire(configurationModel))));
```
Comment (sandy081) [inline src/vs/workbench/services/configuration/browser/configuration.ts]: We should dispose old listener

**Review by sandy081 — CHANGES_REQUESTED** [review]
Feedback provided

**Comment by alexr00** [comment]
@sandy081 is this ready to merge?

**Comment by sandy081** [comment]
Sorry for getting back to this late. Please go ahead and merge it.

**Review by sandy081 — APPROVED** [review]

**Pull request merged**